### ISSUE TITLE
Bug in ScaledFunction.java (!?)

### ISSUE THREAD

**Issue by sbondorf** [NetCal/DiscoSNC#5]
in the method `public double getValue(double theta, Map<Integer, Hoelder> parameters)` there are checks for `original.getParameters().containsKey(hoelder)`.

original is an object of `SymbolicFunction` and `getParameters()` returns `Map<Integer, Hoelder>`, i.e., keys are Integer objects.

Should the code test `containsValue(hoelder)`?

**Comment by sbondorf** [comment]
Spotbugs finds it as well, calls it "No relation between generic parameter and method argument" and categorizes it as
```
Rank: Scariest (1), confidence: High
Pattern: GC_UNRELATED_TYPES 
Type: GC, Category: CORRECTNESS (Correctness)
```

**Comment by sbondorf** [comment]
Fixed in commit d246a0fdee42b9b6cb6b4f954fcbb4c6b8a27bcd.
